### PR TITLE
Issue 6276 - Schema lib389 object is not keeping custom schema data upon editing

### LINKED ISSUE #6276: Schema lib389 object is not keeping custom schema data upon editing
Description of problem:
While editing a custom schema attribute, lib389 and Web UI is not keeping some defined data:

1) if the "Description" is updated, the "OID" is reverted to "<ATTRIBUTE_NAME>-oid"

2) if the "OID" is updated, then the "Description" field shows "No description".

How reproducible:
Always.

Steps to Reproduce:
1. Create a custom attribute ( Schema tab / Attributes tab / "Add Attribute" button )
2. Edit the newly created attribute and change either the description or the OID
3. Save
4. Only the updated field is kept. The non-updated field is reverted to its default value.

Actual results:
Custom datum is lost.

Expected results:
Keep all custom values.

Additional info:
1) A workaround is to update both fields at the same time:

Set the two fields ( Description and OID ) to dummy values and save.
Edit the attribute again
Set the two fields to the correct values and save
2) JS Console:

```
CMD: cmdOperationAttribute: Do the replace operation on Attribute ==> dsconf -j ldapi://%2fvar%2frun%2fslapd-localhost.socket schema attributetypes replace customAttr --syntax 1.3.6.1.4.1.1466.115.121.1.15 --single-value --user-mod --oid 1.2.3.4

CMD: cmdOperationAttribute: Do the replace operation on Attribute ==> dsconf -j ldapi://%2fvar%2frun%2fslapd-localhost.socket schema attributetypes replace customAttr --syntax 1.3.6.1.4.1.1466.115.121.1.15 --single-value --user-mod --desc customAttr_Desc
```

### PR DESCRIPTION
**Pull request by droideck**
Description: Fix the condition in _edit_schema_object function, so it corretcly evaluates when we need to set an empty value to schema object. Add tests to cover add and edit Schema operations.

Fixes: https://github.com/389ds/389-ds-base/issues/6276

Reviewed by: ?

### CODE REVIEW

File: dirsrvtests/tests/suites/clu/schema_test.py [hunk 1]
```diff
@@ -117,25 +98,223 @@ def test_mrs(topo):
...
+    }
+    schema.edit_objectclass(oc_name, edit_parameters)
+
+    # Verify the changes
+    edited_oc = schema.query_objectclass(oc_name, json=True)
+    assert edited_oc['oc']['desc'][0] == 'Updated description for test edit object class'
+    assert set(edited_oc['oc']['must']) == set(['cn', 'sn'])
+    assert set(edited_oc['oc']['may']) == set(['description', 'telephoneNumber'])
+
+    # Clean up
```
Comment (tbordaz) [inline dirsrvtests/tests/suites/clu/schema_test.py:203]: Should it be done in a finalizer ?

File: dirsrvtests/tests/suites/clu/schema_test.py [hunk 2]
```diff
@@ -117,25 +98,223 @@ def test_mrs(topo):
...
+    schema.edit_attributetype(attr_name, edit_parameters)
+
+    # Verify the changes
+    edited_attr = schema.query_attributetype(attr_name, json=True)
+    assert 'sup' not in edited_attr['at'] or not edited_attr['at']['sup']
+    assert 'equality' not in edited_attr['at'] or not edited_attr['at']['equality']
+    assert 'ordering' not in edited_attr['at'] or not edited_attr['at']['ordering']
+    assert 'substr' not in edited_attr['at'] or not edited_attr['at']['substr']
+
+    # Clean up
```
Comment (tbordaz) [inline dirsrvtests/tests/suites/clu/schema_test.py]: finalizer ?

File: dirsrvtests/tests/suites/clu/schema_test.py [hunk 3]
```diff
@@ -117,25 +98,223 @@ def test_mrs(topo):
...
+
+    # Verify that both the description and OID are kept
+    re_edited_attr = schema.query_attributetype(attr_name, json=True)
+    assert re_edited_attr['at']['oid'][0] == new_oid, f"OID not updated, current value: {re_edited_attr['at']['oid'][0]}"
+    assert re_edited_attr['at']['desc'][0] == new_desc, f"Description changed unexpectedly to {re_edited_attr['at']['desc'][0]}"
+
+    # Additional check: Ensure the attribute name wasn't used as the OID
+    assert re_edited_attr['at']['oid'][0] != f"{attr_name}-oid", "OID was incorrectly set to '<ATTRIBUTE_NAME>-oid'"
+
+    # Clean up
```
Comment (tbordaz) [inline dirsrvtests/tests/suites/clu/schema_test.py]: finalizer ?

**Review by tbordaz — APPROVED** [review]
LGTM

**Review by progier389 — APPROVED** [review]
It definitively looks nicer with the parametrized fixture ! -:) 
LGTM for me but you should better wait for the tests completion and check the result before merging.

**Pull request merged**